// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.cmrt.sdk.public_key_service.v1;

option java_multiple_files = true;
option java_outer_classname = "PublicKeyServiceProto";
option java_package = "com.google.cmrt.sdk.public_key_service.v1";

import "cc/core/common/proto/common.proto";
import "google/protobuf/timestamp.proto";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// Service to fetch public keys.
service PublicKeyService {
  // Fetches latest public keys from KeyService.
  rpc ListPublicKeys(ListPublicKeysRequest) returns (ListPublicKeysResponse) {}
}

// Request to list public keys.
message ListPublicKeysRequest {}

// Response of listing public keys
message ListPublicKeysResponse {
  // The execution result.
  scp.core.common.proto.ExecutionResult result = 1;
  // PublicKey list.
  repeated PublicKey public_keys = 2;
  // The expiration time of the public keys.
  google.protobuf.Timestamp expiration_time = 3;
}

// Wrap of public key.
message PublicKey {
  // The id of the public key.
  string key_id = 1;
  // The value of the public key.
  string public_key = 2;
};
